libfwupdplugin/fu-chunk.c

```diff
@@ -438,8 +438,11 @@ fu_chunk_array_new(const guint8 *data,
 		}
 
 		/* cut the packet so it does not straddle multiple blocks */
-		if (page_sz != packet_sz && page_sz > 0)
-			chunksz = MIN(chunksz, (offset + packet_sz) % page_sz);
+		if (page_sz != packet_sz && page_sz > 0) {
+			gsize chunksz_tmp = MIN(chunksz, (offset + packet_sz) % page_sz);
+			if (chunksz_tmp != 0)
+				chunksz = chunksz_tmp;
```

Comment:
can you also add a test to `libfwupdplugin/fu-self-test.c` please? i.e. copy fu_chunk_array_func and add one that this bug would fix. Also, in general commit messages normally start with a capital letter, unless we're using a prefix like `trivial: ` -- thanks. I think a good commit message for this would be something like `Fix a potential hang when creating a chunk array with unusual page sizes` or something like that.